src/components/item/ItemsContainer.js

```diff
@@ -1,13 +1,87 @@
 import React from 'react';
 import { useStore } from 'store/index';
-import Items from 'components/item/Items';
 import FilterOptions from 'components/filter/FilterOptions';
 import Cart from 'components/cart/Cart';
 import Filters from 'components/filter/Filters';
+import Item from 'components/item/Item';
 
 const ItemsContainer = () => {
     const { state, dispatch } = useStore();
-    const { size, cart, products } = state;
+    const {
+      size,
+      cart,
+      products,
+      hasBeenFiltered,
+      currentProducts
+    } = state;
+
+    // Add selected item to customers cart
+    const addToCart = (item, numberOfItems) => {
+      // Check if customer selected the size 
+      if (size.size === null || size.id !== item) {
+          alert('Please select a size for this item');
+          return;
+          // Check if customer selected the number of items
+      } else if (numberOfItems.id !== null && item !== numberOfItems.id) {
+          alert('Please select number of items for the product you would like to add');
+          return;
+      } else {
+          let alreadyInCart = false;
+          const currentCart = cart;
+          // Get the item to add to the cart
+          let itemToAdd = products.find(product => product.product_id === parseInt(item));
+          // Check if that item is already in the cart
+          currentCart.forEach(cart => {
+              if (parseInt(item) === cart.product_id) {
+                  itemToAdd.customerSize = { ...cart.customerSize };
+                  alreadyInCart = true;
+              }
+          });
+          // If not in the cart initalize new selected size and quantity object
+          if (alreadyInCart === false) {
+              itemToAdd.customerSize = { s: 0, m: 0, l: 0 };
+          }
+
+          // Check what size selected and see if;
+          // Item is in stock or
+          // Item has the desired amount in stock 
+          if (size.size === 's') {
+              checkSizeOfItem(itemToAdd, parseInt(numberOfItems.number), 's', alreadyInCart);
+          } else if (size.size === 'm') {
+              checkSizeOfItem(itemToAdd, parseInt(numberOfItems.number), 'm', alreadyInCart);
+          } else if (size.size === 'l') {
+              checkSizeOfItem(itemToAdd, parseInt(numberOfItems.number), 'l', alreadyInCart);
+          }
+          // If item is already in the cart don't add it again
+          if (alreadyInCart === true) {
+              dispatch({ type: 'setCurrentSize', payload: { size: null, id: null } });
+              return;
+          }
+
+          currentCart.push(itemToAdd);
+          dispatch({ type: 'addItemToCart', payload: currentCart });
+          dispatch({ type: 'setCurrentSize', payload: { size: null, id: null } });
+      }
+  };
+
+    let itemsList = [];
+    if (!hasBeenFiltered) {
+      itemsList = products.map((product) => (
```

Comment:
It's not necessary to require three different files to render a list of components, just do your list filtering in the container here